activesupport/lib/active_support/dependencies/autoload.rb

```diff
@@ -26,11 +26,13 @@ module ActiveSupport
   #   MyLib.eager_load!
   module Autoload
     def self.extended(base) # :nodoc:
-      base.class_eval do
-        @_autoloads = {}
-        @_under_path = nil
-        @_at_path = nil
-        @_eager_autoload = false
+      if RUBY_VERSION < "3"
+        base.class_eval do
+          @_autoloads = nil
+          @_under_path = nil
+          @_at_path = nil
+          @_eager_autoload = false
+        end
```

Comment:
Maybe we can do this, but again I thought 2.7 was fine, might not be worth it.

Getting rid of the `class_eval` was just the cherry on top, the main thing I wanted here to clear the `autoloads` hash.